yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/main.nr

```diff
@@ -231,6 +241,102 @@ contract NonNativeToken {
         context.finish(inputs)
     } 
 
+    // Shield creates a way for a user to move tokens from the public context into the private context.
+    open fn shield(
+        inputs: PublicContextInputs,
+        amount: Field,
+        secretHash: Field,
+        _padding: [Field; abi::MAX_ARGS - 2]
+    ) {
+        // Create a commitment to the amount
```

Comment:
Shouldn't we ensure that `amount <= public_balances[msg.sender]`?

Reply:
That would leak the sender, also the amount is being minted from private, the hope is that this function would only be callable by a private function where this has been constrained in private